plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakePartitionsTable.java

```diff
@@ -0,0 +1,472 @@
+                        column.getBaseColumnName(),
+                        RowType.from(ImmutableList.of(
+                                new RowType.Field(Optional.of("min"), column.getType()),
+                                new RowType.Field(Optional.of("max"), column.getType()),
+                                new RowType.Field(Optional.of("null_count"), BIGINT)))))
+                .collect(toImmutableList());
+        if (metricColumns.isEmpty()) {
+            return Optional.empty();
+        }
+        return Optional.of(RowType.from(metricColumns));
+    }
+
+    private static boolean isPrimitiveType(Class<?> type)
+    {
+        return (type == long.class || type == double.class || type == boolean.class || type == byte.class || type == int.class);
+    }
+
+    private record DeltaLakePartitionStatistics(long fileCount, long size, Map<String, Object> minValues, Map<String, Object> maxValues, Map<String, Long> nullCounts)
+    {
+        public static class Builder
+        {
+            private final List<DeltaLakeColumnHandle> columns;
+            private final TypeManager typeManager;
+            private long fileCount;
+            private long size;
+            private final Map<String, ColumnStatistics> columnStatistics = new HashMap<>();
+            private final Map<String, Optional<Long>> nullCounts = new HashMap<>();
+            private boolean ignoreDataColumn;
+
+            public Builder(List<DeltaLakeColumnHandle> columns, TypeManager typeManager)
+            {
+                this.columns = ImmutableList.copyOf(requireNonNull(columns, "columns is null"));
+                this.typeManager = requireNonNull(typeManager, "typeManager is null");
+            }
+
+            public void acceptAddFileEntry(AddFileEntry addFileEntry)
+            {
+                if (addFileEntry.getDeletionVector().isPresent()) {
+                    return;
+                }
+
+                fileCount++;
+                size += addFileEntry.getSize();
+
+                if (addFileEntry.getStats().isEmpty()) {
+                    columnStatistics.clear();
+                    nullCounts.clear();
+                    ignoreDataColumn = true;
+                }
+
+                if (ignoreDataColumn) {
+                    return;
+                }
+
+                addFileEntry.getStats().ifPresent(stats -> {
+                    for (DeltaLakeColumnHandle column : columns) {
+                        updateMinMaxStats(
+                                column.getBaseColumnName(),
+                                column.getType(),
+                                stats.getMinColumnValue(column).orElse(null),
+                                stats.getMaxColumnValue(column).orElse(null),
+                                Optional.of(0L),
+                                stats.getNumRecords().orElse(0L));
+                        updateNullCountStats(column.getBaseColumnName(), Optional.of(0L));
```

Comment:
```suggestion
updateNullCountStats(column.getBaseColumnName(), stats.getNullCount(column.getBaseColumnName()));
```

Please cover the functionality with specific assertions in `io.trino.plugin.deltalake.TestDeltaLakeSystemTables#testPartitionsTable` 

Be sure to insert also `NULL` for the `_bigint` column

Reply:
Added to `io.trino.plugin.deltalake.TestDeltaLakeSystemTables#testPartitionsTable`